usrsctplib/netinet/sctp_os_userspace.h

```diff
@@ -285,11 +285,11 @@ typedef char* caddr_t;
 #include <sys/socket.h>
 #if defined(__DragonFly__) || defined(__FreeBSD__) || defined(__linux__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__native_client__) || defined(__Fuchsia__) || defined(__EMSCRIPTEN_PTHREADS__)
```

Comment:
With regards to the errors on macOS, it looks like the `#if` line should also do `|| defined(__APPLE__)` since it is fine to `#include <pthread.h>` on Apple platforms.

It looks like on Apple platforms, `<pthread.h>` is transitively included before reaching this part of the file (to be 100% we should look the preprocessed output) so even if the `#include` gets removed, the `typedef`s work.